libsolidity/ast/AST.cpp

```diff
@@ -136,27 +139,39 @@ FunctionDefinition const* ContractDefinition::receiveFunction() const
 	return nullptr;
 }
 
-vector<EventDefinition const*> const& ContractDefinition::interfaceEvents() const
+vector<EventDefinition const*> const& ContractDefinition::interfaceEvents(bool _ignoreErrors) const
 {
 	return m_interfaceEvents.init([&]{
+		vector<EventDefinition const*> unfilteredEvents;
+		for (ContractDefinition const* contract: annotation().linearizedBaseContracts)
+			unfilteredEvents += contract->events();
+		solAssert(annotation().creationCallGraph.set() == annotation().deployedCallGraph.set(), "");
+		if (annotation().creationCallGraph.set())
+		{
+			unfilteredEvents += (*annotation().creationCallGraph)->emittedEvents;
+			unfilteredEvents += (*annotation().deployedCallGraph)->emittedEvents;
+		}
+		else
+			solAssert(_ignoreErrors, "Interface events requested before call graph has been computed.");
+
 		set<string> eventsSeen;
 		vector<EventDefinition const*> interfaceEvents;
-
-		for (ContractDefinition const* contract: annotation().linearizedBaseContracts)
-			for (EventDefinition const* e: contract->events())
+		for (EventDefinition const* e: unfilteredEvents)
+		{
+			/// NOTE: this requires the "internal" version of an Event,
+			///       though here internal strictly refers to visibility,
+			///       and not to function encoding (jump vs. call)
+			FunctionType const* function = e->functionType(true);
+			solAssert(function, "");
+			if (!function->interfaceFunctionType() && _ignoreErrors)
+				continue;
+			string eventSignature = function->externalSignature();
```

Comment:
I think this would cause issues when in the following case:

```
  library L {
	  event e1(uint b);
  }

  function f() {
	  emit L.e1(5);
  }

  contract C {
	  event e1(uint indexed a);
	  function g() public {
		  f();
	  }
  }
```

Because the signature does not contain "indexed".

The ABI for this test case is:

```
[
  {
    "anonymous": false,
    "inputs": [
      {
        "indexed": false,
        "internalType": "uint256",
        "name": "a",
        "type": "uint256"
      }
    ],
    "name": "e1",
    "type": "event"
  },
  {
    "inputs": [],
    "name": "g",
    "outputs": [],
    "stateMutability": "nonpayable",
    "type": "function"
  }
]
```
Which doesn't contain the library event.

Reply:
We have the same problem during inheritance. Let me see what I can do.

Reply:
I actually don't know why we do this filtering at all. Events can be overloaded but defining two events with the same name and parameter types in an inheritance hierarchy is an error anyway.